commands/create.go

```diff
@@ -0,0 +1,189 @@
+		return errors.New("You are not logged in. Use '" + config.ProgramName + " login' to log in.")
+	}
+	if cmdClusterID == "" {
+		// use default cluster if possible
+		clusterID, _ := config.GetDefaultCluster()
+		if clusterID != "" {
+			cmdClusterID = clusterID
+		} else {
+			return errors.New("No cluster given. Please use the -c/--cluster flag to set a cluster ID.")
+		}
+	}
+	return nil
+}
+
+// createKubeconfig adds configuration for kubectl
+func createKubeconfig(cmd *cobra.Command, args []string) {
+	client := gsclientgen.NewDefaultApi()
+	authHeader := "giantswarm " + config.Config.Token
+
+	// parameters given by the user
+	ttlHours := int32(cmdTTLDays * 24)
+	if cmdDescription == "" {
+		cmdDescription = "Added by user " + config.Config.Email + " using 'g8m create kubeconfig'"
+	}
+
+	addKeyPairBody := gsclientgen.AddKeyPairBody{Description: cmdDescription, TtlHours: ttlHours}
+
+	fmt.Println("Creating new key-pair…")
+
+	keypairResponse, _, err := client.AddKeyPair(authHeader, cmdClusterID, addKeyPairBody)
+
+	if err != nil {
+		fmt.Println(color.RedString("Error in createKubeconfig:"))
+		log.Fatal(err)
+		fmt.Println("keypairResponse:", keypairResponse)
+		fmt.Println("addKeyPairBody:", addKeyPairBody)
+		os.Exit(1)
+	}
+
+	if keypairResponse.StatusCode == 10000 {
+		msg := fmt.Sprintf("New key-pair created with ID %s and expiry of %v hours",
+			util.Truncate(util.CleanKeypairID(keypairResponse.Data.Id), 10),
+			ttlHours)
+		fmt.Println(msg)
+
+		// store credentials to file
+		caCertPath := util.StoreCaCertificate(config.ConfigDirPath, cmdClusterID, keypairResponse.Data.CertificateAuthorityData)
+
+		clientCertPath := util.StoreClientCertificate(config.ConfigDirPath, cmdClusterID, keypairResponse.Data.Id, keypairResponse.Data.ClientCertificateData)
+
+		clientKeyPath := util.StoreClientKey(config.ConfigDirPath, cmdClusterID, keypairResponse.Data.Id, keypairResponse.Data.ClientKeyData)
+
+		fmt.Println("Certificate and key files written to:")
+		fmt.Println(caCertPath)
+		fmt.Println(clientCertPath)
+		fmt.Println(clientKeyPath)
+
+		// TODO: Take this from the cluster object
+		apiEndpoint := "https://api." + cmdClusterID + ".k8s.gigantic.io"
+
+		// edit kubectl config
+		util.KubectlSetCluster(cmdClusterID, apiEndpoint, caCertPath)
+		util.KubectlSetCredentials(cmdClusterID, clientKeyPath, clientCertPath)
+		util.KubectlSetContext(cmdClusterID)
+		util.KubectlUseContext(cmdClusterID)
+
+		fmt.Printf("Switched to kubectl context 'giantswarm-%s'\n\n", cmdClusterID)
+
+		// final success message
+		color.Green("kubectl is set up. Check it using this command:\n\n")
+		color.Yellow("    kubectl cluster-info\n\n")
+		color.Green("Whenever you want to switch to using this context:\n\n")
+		color.Yellow("    kubectl config set-context giantswarm-%s\n\n", cmdClusterID)
+
+	} else {
+		fmt.Printf("Unhandled response code: %v", keypairResponse.StatusCode)
+		fmt.Printf("Status text: %v", keypairResponse.StatusText)
```

Comment:
I find it weird to use different libraries to do one thing. You print information using `fmt` and `color`. There should only be one way to do it.